mainsite/selling/forms.py

```diff
@@ -44,20 +90,35 @@ def save(self, commit=True):
 
         if commit:
             item.save()
+
+            # we have to be careful about order here, because of the
+            #  unique_together constraint and sqlite doesn't allow deferred
+            for pic in self.otherPictures.all():
+                pic.delete()    # clear out taken positions
+            for pic in self.specialPictures:
+                pic.position += 2*len(self.cleaned_data["pictures"])
+                pic.save()  # move them up to guaranteed-untaken positions
+            for pic in self.specialPictures:
+                pic.position -= 2*len(self.cleaned_data["pictures"])
+                pic.save()      # move still-around ones to new positions
+            for pic in self.cleaned_data["pictures"]:
+                pic.save()      # save everything
+
         return item
 
-    def clean_item_picture(self):
-        pic = self.cleaned_data['item_picture']
+    def _clean_a_picture(self, pic):
+        #pic = self.cleaned_data['curr_picture']
         if pic.size > settings.MAX_UPLOAD_SIZE:
             raise forms.ValidationError(_('Filesize is too large and image could not be automatically downsized: Please use a smaller or lower-resolution image. Maximum file size is: %(max_size).1f %(type)s'),
             params={'max_size': 1024**(math.log(settings.MAX_UPLOAD_SIZE, 1024)%1), 'type': ["B", "KB", "MB", "GB", "TB"][int(math.floor(math.log(settings.MAX_UPLOAD_SIZE, 1024)))] }, code='toolarge')
 
         im = Image.open(pic)
-
         if im.format.lower() not in settings.ALLOWED_UPLOAD_IMAGES:
             raise forms.ValidationError(_("Unsupported file format. Supported formats are %s."
                                           % ", ".join(settings.ALLOWED_UPLOAD_IMAGES)))
 
+        if pic.name == "/static/mainsite/images/imagenotfound.png": # TODO change this to use a back lookup
```

Comment:
When would this happen, and what do you mean by a back lookup?

Reply:
It shouldn't ever happen, because the file upload fields for newly-user-uploaded images are marked required. But I didn't realize I was going to do that when I wrote the code, so I had backup added here. (And the 'back lookup' I just meant using whatever the equivalent of the {% static %} tag would be on serverside so it isn't hardcoded -- I didn't want to bother googling it at the time tbh)
I've removed the line, it should be extraneous